public/request/game/generate-forum-topic.php

```diff
@@ -12,8 +13,12 @@
     'game' => 'required|integer|exists:GameData,ID',
 ]);
 
-if (generateGameForumTopic($user, (int) $input['game'], $forumTopicID)) {
-    return redirect(url("/viewtopic.php?t=$forumTopicID"))->with('success', __('legacy.success.create'));
+$userModel = User::firstWhere('User', $user);
```

Comment:
I've been assuming anything validated by `authenticateFromCookie` is just `request()->user()`. Is that a bad assumption?

Reply:
It's a safe assumption based on `authenticateFromCookie`'s implementation details:

```php
/** @var ?User $user */
$user = auth()->user();
```

`$userDetailsOut` is just:

```php
$userDetailsOut = $user->toArray();
```

Eventually we'll remove `authenticateFromCookie()` completely, but a good step forward might be dropping that `->toArray()` call so we don't have to instantiate a `$userModel` value all over the place.

Reply:
Related: https://github.com/RetroAchievements/RAWeb/pull/2404